app/src/main/java/io/stormbird/wallet/service/KeyService.java

```diff
@@ -399,15 +394,14 @@ private void getAuthenticationForSignature()
         {
             KeyStore keyStore = KeyStore.getInstance(ANDROID_KEY_STORE);
             keyStore.load(null);
-            SecretKey secretKey = (SecretKey) keyStore.getKey(currentKeyAddress, null);
-            String encryptedHDKeyPath = getEncryptedFilePath(currentKeyAddress);//getFilePath(context, currentKey + "hd");
-            boolean fileExists = new File(encryptedHDKeyPath).exists();
-            if (!fileExists || secretKey == null)
+            SecretKey secretKey = (SecretKey) keyStore.getKey(currentWallet.address, null);
+            String encryptedHDKeyPath = getFilePath(context, currentWallet.address);
+            if (!new File(encryptedHDKeyPath).exists() || secretKey == null)
             {
                 signCallback.GotAuthorisation(false);
                 return;
             }
-            byte[] iv = readBytesFromFile(getFilePath(context, currentKeyAddress + "iv"));
+            byte[] iv = readBytesFromFile(getFilePath(context, currentWallet.address + "iv"));
```

Comment:
There might still be a slight issue by using the address of a wallet as a file identifier as it could allow someone with renaming access to the file system to switch around two wallets. Of course transactions would only go through if the same mnemonic was used for both wallets. Thus this seems to be a very minor issue.